graspologic/layouts/auto.py

```diff
@@ -249,6 +245,27 @@ def _node2vec_for_layout(
     return graph, tensors, labels
 
 
+def _to_undirected(graph: nx.DiGraph) -> nx.Graph:
+    sym_g = nx.Graph()
+    weighted = nx.is_weighted(graph)
```

Comment:
This is still somewhat problematic - turns out that nx.is_weighted doesn't actually test each weight value to see if it is a number or not, which leads to a valueerror down the line if someone does `weight='i am a donut'`.  I just merged a function into graspologic.preconditions to check each value to make sure they're either all weighted and numeric or not weighted or not numeric.

Reply:
One of the strategies I took instead was to say "if it isn't fully weighted with numeric weights then I'm just going to treat it as an unweighted graph and provide my own weight of 1 for each edge", with a warning

Reply:
I see, I'll use the `is_real_weighted` instead of the networkx 'is_weighted' function. Right now if the graph isn't fully weighted or unweighted I just throw an error - should I add a weight of 1 for every edge (even the already weighted ones) and make it a warning instead?

Reply:
im against adding 1-weight edges, personally, unless the entire graph is unweighted (in which case I don't think it's necessary for leiden anyway)

Reply:
We could also make the graph unweighted and remove the weights if only some of the edges are weighted?

Reply:
@daxpryce in my head the code should run only when all edges or no edges are weighted, and if they are mixed I'd rather see the code just error. i dont want to assume anything about what they meant. thoughts?

Reply:
@daxpryce seems to agree with me on "throw an error"

Reply:
@bdpedigo I completely agree.  If it's fully unweighted we can probably safely presume an edge weight of 1 (may still be wrong, but I don't feel like that's our responsibility).  If it's fully weighted, then it's fully weighted.  If it's partially weighted, we have no idea what the user would want us to do, so I say throw a `ValueError("Fully weight your graph if you want us to do anything with it because we are not mindreaders kthx")`

Reply:
Please sound more professional than I do, though